app/serializers/toggle_flag.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+class ToggleFlag
+  include Service::Base
+
+  contract
+  model :flag
+  policy :invalid_access
+
+  transaction do
+    step :toggle
+    step :log
+  end
+
+  class Contract
+    attribute :flag_id, :integer
```

Comment:
You could add a validates presence for the `flag_id` too

https://github.com/discourse/discourse/blob/d1191b7f5fbe9e8ea1b645d5a61c79fa93693c0c/plugins/chat/app/services/chat/create_thread.rb#L38-L38